src/platform/plugins/shared/console/server/lib/spec_definitions/js/retriever.ts

```diff
@@ -105,6 +105,28 @@ export const retriever = (specService: SpecDefinitionsService) => {
       match_criteria: {},
       rank_window_size: 10,
     },
+    pinned: {
+      __template: {
+        retriever: {},
+        ids: [],
+        docs: [],
+        match_criteria: {},
+      },
+      retriever: {
+        __scope_link: '.',
+      },
+      // Only one of 'ids' or 'docs' should be used at a time
+      ids: {
+        __description:
+          "Specify an array of document IDs to pin. Only one of 'ids' or 'docs' can be used.",
```

Comment:
never seen `__description` being used before - is this a pattern we use somewhere else?

there's a problem here - we autocomplete here with an object and not an array:

<img width="242" alt="Screenshot 2025-06-13 at 09 40 53" src="https://github.com/user-attachments/assets/3296abe2-770b-41ee-8650-5b00dbcd17e4" />

<img width="252" alt="Screenshot 2025-06-13 at 09 40 59" src="https://github.com/user-attachments/assets/9fd0caa9-23a7-44e9-9bde-559fc92c0cd0" />